x/stakeibc/keeper/msg_server.go

```diff
@@ -112,6 +112,14 @@ func (k msgServer) RegisterHostZone(goCtx context.Context, msg *types.MsgRegiste
 		return nil, errorsmod.Wrapf(err, "unable to create community pool redeem account for host zone %s", chainId)
 	}
 
+	// Validate the community pool treasury address if it's non-empty
+	if msg.CommunityPoolTreasuryAddress != "" {
+		_, err := utils.AccAddressFromBech32(msg.CommunityPoolTreasuryAddress, msg.Bech32Prefix)
```

Comment:
Just verifying, this function doesn't error on modular accounts, ICA accounts, or CW accounts? I think that's the case, but just want to double-check!

Reply:
yup that's right! it works for all of them